src/TypedMessageManager.test.ts

```diff
@@ -0,0 +1,332 @@
+import TypedMessageManager from './TypedMessageManager';
+const typedMessage = [
+	{
+		name: 'Message',
+		type: 'string',
+		value: 'Hi, Alice!'
+	},
+	{
+		name: 'A number',
+		type: 'uint32',
+		value: '1337'
+	}
+];
```

Comment:
Is it worth to add a test for a message with data that has >= 2 levels deep ?

Reply:
Just making sure that works for complex objects

Reply:
The way I understand it, these managers only have the responsibility to handle messages flow and not the data itself, that would be a responsibility for `eth-sig-util` in this case. But I can be wrong and we could add more complex data structures extending from `object[]`. For that reason there are also tests for `eth_signTypedData_v3` with typed messages validation according to the EIP712.